collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -207,6 +208,10 @@ private Config loadConfig(Map<String, Object> yamlConfig) throws MalformedObject
           }
         }
 
+        if (yamlConfig.containsKey("collectorNamePattern")) {
+            cfg.collectorNamePattern = (String)yamlConfig.get("collectorNamePattern");
```

Comment:
Please validate if collectorNamePattern is a valid regex, so that `jmx_exporter` fails immediately on config error. Otherwise it starts successfully but fails when the Prometheus server scrapes.

Reply:
Added validation check as well as unit test to verify coverage.